experiment/tracer/static/script.js

```diff
@@ -0,0 +1,273 @@
+'use strict';
+
+var logger = {
+  data: [],
+  message: '',
+  options: {
+    request: {
+      endpoint: 'https://' + window.location.hostname + '/trace?',
+      params: '',
+      config: {}
+    },
+    messages: {
+      invalidInput: 'Invalid input. Please pass a link from a PR or PR comment and profit',
+      invalidHost: 'Invalid host. Host needs to be github.com',
+      invalidParams: 'Invalid params in the URL. Need either "pr", "repo", and "org", or  "issuecomment" to be specified',
+      invalidUrl: 'Invalid link provided. Failed to construct \'URL\'',
+      requestError: 'Fetch error: Status is not 2xx',
+      requestLoading: 'Loading..',
+      emptyLogs: 'No logs to display'
+    }
+  },
+
+  // the actual fetch
+  request: function(endpoint) {
+    generateView.toggleLoader(true);
+    fetch(endpoint).then(
+        function(response) {
+
+          if (response.status !== 200) {
+            generateView.toggleLoader(false);
+            controller.setMessage(logger.options.messages.requestError);
+            return;
+          }
+
+          response.json().then(function(data) {
+            generateView.toggleLoader(false);
+            controller.setData(data);
+          });
+        }
+    ).catch(function(err) {
+      console.log('Fetch Error', err);
+      generateView.toggleLoader(false);
+      controller.setMessage(err);
+    });
+
+  }
+};
+
+var controller = {
+
+  init: function() {
+    generateView.init();
+  },
+
+  // Getter & Setters
+
+  setParams: function(params) {
+    logger.options.request.params = params;
+  },
+
+  setData: function(data) {
+    logger.data = data;
+    // render this view (update the DOM elements with the updated values)
+    generateView.render();
+  },
+
+  getParams: function() {
+    return logger.options.request.endpoint + encodeURI(logger.options.request.params);
+  },
+
+  setMessage: function(msg) {
+    logger.message = msg;
+    generateView.updateMsg();
+  },
+
+  getMessage: function() {
+    return logger.message;
+  },
+
+  // fetch data
+  requestLogs: function() {
+    //  params needed
+    return logger.request(this.getParams());
+  },
+
+  buildParams: function(userValue) {
+
+    if (userValue.length <= 1) {
+      this.setMessage(logger.options.messages.invalidInput);
+      return;
+    }
+
+    try {
+      var url = new URL(userValue);
+    } catch(err) {
+      this.setMessage(logger.options.messages.invalidUrl);
+      return;
+    }
+
+    var host = url.host,
+        urlParser = url.pathname.split('/');
+
+    if (host !== 'github.com') {
+      this.setMessage(logger.options.messages.invalidHost);
+      return;
+    }
+
+    if (urlParser.length > 5) {
+      this.setMessage(logger.options.messages.invalidParams);
+      return;
+    }
+
+    var org = urlParser[1],
+        repo = urlParser[2],
+        pr = urlParser[4];
+
+    var params = 'org=' + org + '&repo=' + repo + '&pr=' + pr;
```

Comment:
Is there a simple lib to encode the URL that would handle formatting and escaping here instead of using string concat?

Reply:
I use encoding when I get the params:
https://github.com/kubernetes/test-infra/pull/6241/files#diff-7d200cf223d93ff3baf058b95c2106feR68. Hm, maybe it could be better to do it at this point.